src/commands/typescript/update.ts

```diff
@@ -67,37 +111,83 @@ export default class Update extends SfdxCommand {
     await fs.writeJson(tsConfigPath, tsConfig);
   }
 
-  private async updateTsVersion(): Promise<void> {
-    const newVersion = this.determineNextTsVersion();
-    const pkg = await Package.create(path.resolve('.'));
+  private async updateEsTarget(): Promise<void> {
+    if (this.isLernaRepo()) {
+      const packagePaths = await this.getPackagePaths();
+      for (const packagePath of packagePaths) {
+        await this.updateEsTargetConfig(packagePath);
+      }
+    } else {
+      await this.updateEsTargetConfig('.');
+    }
+    return;
+  }
 
+  private async updatePackage(packagePath: string, npmPackage: NpmPackage): Promise<void> {
+    const newVersion = this.determineNextTsVersion(npmPackage);
+    const pkg = await Package.create(path.resolve(packagePath));
     if (pkg.packageJson.devDependencies['typescript']) {
       this.ux.log(`Updating typescript version to ${newVersion}`);
       pkg.packageJson.devDependencies['typescript'] = newVersion;
       pkg.packageJson.devDependencies['@typescript-eslint/eslint-plugin'] = 'latest';
       pkg.packageJson.devDependencies['@typescript-eslint/parser'] = 'latest';
     }
-
     // If the prepare script runs sf-install, the install will fail because the typescript version
     // won't match the expected version. So in that case, we delete the prepare script so that we
     // get a successful install
     if (pkg.packageJson.scripts['prepare'] === 'sf-install') {
       delete pkg.packageJson.scripts['prepare'];
     }
+    return pkg.writePackageJson();
+  }
 
-    pkg.writePackageJson();
+  private async updateTsVersion(): Promise<void> {
+    if (this.isLernaRepo()) {
+      const packagePaths = await this.getPackagePaths();
+      const runner = exec('npm view typescript --json', { silent: true });
+      if (runner.code !== 0) {
+        throw new SfdxError('Could not find typescript on the npm registry', 'TypescriptNotFound');
+      }
+      for (const packagePath of packagePaths) {
+        const npmPackage = JSON.parse(runner.stdout) as NpmPackage;
+        await this.updatePackage(packagePath, npmPackage);
+      }
+      return;
+    }
+
+    this.typescriptPkgs.map(async (typescriptPkg) => {
+      this.ux.log(typescriptPkg.version);
+      await this.updatePackage(path.resolve('.'), typescriptPkg);
+    });
   }
 
-  private determineNextTsVersion(): string {
+  private determineNextTsVersion(typescriptPkg: NpmPackage): string {
     return this.flags.version === 'latest' || !this.flags.version
-      ? getString(this.typescriptPkg, 'dist-tags.latest')
+      ? getString(typescriptPkg, 'dist-tags.latest')
       : (this.flags.version as string);
   }
 
-  private retrieveTsPackage(): NpmPackage {
-    const result = exec('npm view typescript --json', { silent: true });
-    if (result.code === 0) {
-      return JSON.parse(result.stdout) as NpmPackage;
+  private async retrieveTsPackages(): Promise<NpmPackage[]> {
+    // Process lerna repos
+    if (this.isLernaRepo()) {
```

Comment:
You don't need to iterate over the packages here either. This method simply gets the version of typescript from `npm view typescript`. So you only need to get it once